tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11882,6 +11909,33 @@ bool hlsl::ShouldSkipNRVO(clang::Sema &sema, clang::QualType returnType,
   return false;
 }
 
+bool hlsl::HasLongVecs(const QualType &qt) {
+  if (qt.isNull()) {
+    return false;
+  }
+
+  if (IsHLSLVecType(qt)) {
+    if (GetHLSLVecSize(qt) > 4)
```

Comment:
>4 [](http://example.com/codeflow?start=29&length=1)

Would this be better defined as a constant somewhere?

Reply:
I agree we should reduce the number of places we put the magic value. Can we make this a language option though? Then we can query the Language Options for something like `maxSupportedVectorSize()`?